lib/Jit/LLILCJit.cpp

```diff
@@ -290,6 +291,18 @@ CorJitResult LLILCJit::compileMethod(ICorJitInfo *JitInfo,
 
         Passes.add(createRewriteStatepointsForGCPass(false));
         Passes.run(*M);
+
+        // PlaceSafepoints phase inlines the contents of gc.safepointPoll()
+        // at all polling locations. So, once the phase has run, remove the
+        // @gc.safepointPoll() function from the module.
+        Function *SafepointPoll = Context.SafepointPoll;
+#if !defined(NDEBUG)
```

Comment:
This assertion (that no references to SafepointPoll remain after PlaceSafepoint phase) failed for some of the new GC tests (not yet added in the CoreCLR test tree). 

The correct way to remove the unreachable functions is to invoke the Inliner (Inliner::removeDeadFunctions()).
However, we shouldn't have to run this optimization in /Od builds.

In order to get the lab job (InsertStatepoints nightly) green, I'll add a workaround to remove trivially unreachable SafepointPolls.